WcaOnRails/app/views/competitions/_competition_form.html.erb

```diff
@@ -261,87 +261,51 @@
         <%= f.input :base_entry_fee_lowest_denomination, as: :money_amount, currency: @competition.currency_code, disabled: disable_money_input, currency_selector: "#competition_currency_code" %>
 
         <script>
-          var baseFeeUSD = -1;
-
-          // Note: unfortunately we cannot trigger a change on the currency
-          // code update, because we don't have access to the currency
-          // "subunit_to_unit" from the js world.
           function UpdateDues(){
-            <% countryBand = CountryBand.where(iso2: @competition.country_iso2).map(&:number)[0] %>
-            <% if countryBand.present? && countryBand > 0 %>
-              var bandFeeUSD = <%= CountryBand::BANDS[countryBand][:value] %>;
-            <% else %>
-              var bandFeeUSD = 0;
-            <% end %>
-            var pricePerCompetitor = Math.max(bandFeeUSD, baseFeeUSD) * 100;
-            var compLimit = $('select[name="competition[competitor_limit_enabled]"]').val() === "true"
-              ? $('input[name="competition[competitor_limit]"]').val() : null;
-            window.wca.cancelPendingAjaxAndAjax('currency_convert', {
-              url: '<%= currency_convert_path %>',
+
+            var compLimitEnabled = $('select[name="competition[competitor_limit_enabled]"]').val();
+            var compLimit = compLimitEnabled === "true" ? $('input[name="competition[competitor_limit]"]').val() : null;
+            var countryId = $('select[name="competition[countryId]"]').val()
+            var currencyCode = $('select[name="competition[currency_code]"]').val()
+            var entryFee = $('input[name="competition[base_entry_fee_lowest_denomination]"]').val();
```

Comment:
Please take a look at how the big code block towards the top of the file handles serialization in the `handleAdjacentCompetitionAjax` method.

In particular, you can encode the entire form via `var obj = $("#competition-form").serializeJSON();` and then only access the parts of the encoded object that you need (again, see above for reference). Saves a ton of `$(selector).val()` calls.